frontend/catalyst/debug/instruments.py

```diff
@@ -32,14 +32,32 @@
 ## API ##
 @contextmanager
 def instrumentation(session_name, filename=None, detailed=False):
-    """Start an instrumentation session. A session cannot be tied to the creation of a QJIT object
+    """Instrumentation session to output information on wall time, CPU time,
+    and intermediate program size of a program during compilation and execution.
+
+    A session cannot be tied to the creation of a QJIT object
     nor its lifetime, because it involves both compile time and runtime measurements. It cannot
     be a context-free process either since we need to write results to an existing results file.
 
     Args:
-        session_name (str): identifier to distinguish multiple results, primarily for humans
-        filename (str): desired path to write results to in YAML format
-        detailed (bool): whether to instrument finegrained steps in the compiler and runtime
+        session_name (str): identifier to distinguish multiple sessions or runs within the same result file
+        filename (str): Desired path to write results to in YAML format. If ``None``, the
+            results will instead be printed to the console.
+        detailed (bool): Whether to instrument fine-grained steps in the compiler and runtime.
+            If ``False``, only high-level steps such as "program capture" and
+            "compilation" are reported.
+
+    **Example**
+
+    >>> @qjit
+    ... def expensive_function(a, b):
+    ...     return a + b
+    >>> with debug.instrumentation("session_name", detailed=False):
+    >>>     expensive_function(1, 2)
+    [DIAGNOSTICS] Running capture                   walltime: 3.299 ms      cputime: 3.294 ms       programsize: 0 lines
+    [DIAGNOSTICS] Running generate_ir               walltime: 4.228 ms      cputime: 4.225 ms       programsize: 14 lines
+    [DIAGNOSTICS] Running compile                   walltime: 57.182 ms     cputime: 12.109 ms      programsize: 121 lines
+    [DIAGNOSTICS] Running run                       walltime: 1.075 ms      cputime: 1.072 ms
     """
     session = InstrumentSession(session_name, filename, detailed)
```

Comment:
```suggestion
    """
    #     A session cannot be tied to the creation of a QJIT object
    # nor its lifetime, because it involves both compile time and runtime measurements. It cannot
    # be a context-free process either since we need to write results to an existing results file.
    session = InstrumentSession(session_name, filename, detailed)
```